secvarctl.c

```diff
@@ -112,22 +112,29 @@ int main(int argc, char *argv[])
 static struct backend *getBackend()
 {
 	char *buff = NULL, *secVarFormatLocation = "/sys/firmware/secvar/format";
-	size_t buffSize;
+	size_t buffSize, max_buff_size;
 	struct backend *result = NULL;
+	int i;
 	// if file doesnt exist then print warning and keep going
 	if (isFile(secVarFormatLocation)) {
 		prlog(PR_WARNING, "WARNING!! Platform does not support secure variables\n");
 		goto out;
 	}
-	buff = getDataFromFile(secVarFormatLocation, &buffSize);
+	// get max size of backend name
+	max_buff_size = strlen(backends[0].name);
```

Comment:
Perhaps use a constant/macro for this in the header (or wherever `backends[]` is defined), or something else that can be done at compile-time. This is going to have to be updated every time a new backend is added, with a potentially longer name.

Personally, I'd suggest defining a reasonable maximum in a macro constant next to the table, maybe ~64 bytes or so. If we end up with a longer backend name, that number can be incremented at the same time.

Reply:
The alternative is to also loop through the backends first to get the maximum format name length, would also prevent needing to ever adjust this code in the future. Bit excessive though.

Reply:
Ah that's what I get for speedreading.

Hmmm... this is fine then, ignore what I said. There's probably a way to do this fully in preprocessor logic, but that's a needless optimization.